app/src/main/java/com/example/taskyy/data/repositories/AuthRepositoryImpl.kt

```diff
@@ -14,38 +15,43 @@ import kotlinx.coroutines.GlobalScope
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.runBlocking
 import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.await
+import retrofit2.awaitResponse
 import javax.inject.Inject
+import kotlin.math.log
 
 class AuthRepositoryImpl @Inject constructor(
     private val userDao: UserDao,
     private val retrofit: TaskyyApi,
 ): AuthRepository {
 
     override fun addUserToDatabase(user: User) {
-        runBlocking { launch {
-            userDao.insertUser(user.mapToUserEntity())
-             }
+        runBlocking {
+            launch {
+                userDao.insertUser(user.mapToUserEntity())
+            }
         }
     }
```

Comment:
No runBlocking here as this can block your app's UI. If you have a suspend function like `userDao.insertUser` which you'd like to call, you have two options:
1. Call it in a launched coroutine (e.g. `viewModelScope`, `GlobalScope` or similar)
2. Call it in another suspend function

In the last PR I mentioned that I almost always recommend to expose suspend functions to your ViewModel and then execute them in `viewModelScope.launch` there. So you would need to change all your repo functions to `suspend fun`:
```
// Also add this suspend keyword to the AuthRepository interface
override suspend fun addUserToDatabase(user: User) {
    // Here you can now execute other suspend functions without runBlocking or a scope
}
```